File: mpi1/send.c
#include <stdio.h>
#include <mpi.h>
 
int main(int argc, char *argv[]) {
   int          rank,num,tag;
   MPI_Status   status;
 
   MPI_Init(&argc, &argv);
   MPI_Comm_rank(MPI_COMM_WORLD,&rank);
 
   tag=88;
 
   if (rank==0) {
      num=10;
      MPI_Send(&num,1,MPI_INT,1,tag,MPI_COMM_WORLD);
      printf("Process %d sent the number %d\n",rank,num);
   }
 
   if (rank==1) {
      MPI_Recv(&num,1,MPI_INT,0,tag,MPI_COMM_WORLD,&status);
      printf("Process %d received the number %d\n",rank,num);
   }
   
   MPI_Finalize();
   return 0;
}

File: mpi1/send.f90
program send
  implicit none
  include 'mpif.h'
  integer :: rank, ierr
  integer :: num, tag, stat(MPI_STATUS_SIZE)

  call MPI_Init(ierr)
  call MPI_Comm_rank(MPI_COMM_WORLD,rank,ierr)

  tag=88

  if (rank == 0) then
    num=10
    call MPI_Send(num,1,MPI_INTEGER,1,tag,MPI_COMM_WORLD,ierr)
    print*, 'Process',rank,'sent the number',num
  end if

  if (rank == 1) then
    call MPI_Recv(num,1,MPI_INTEGER,0,tag,MPI_COMM_WORLD,stat,ierr)
    print*, 'Process',rank,'received the number',num
  end if

  call MPI_Finalize(ierr)
end program send
